lib/scsi/lun.c

```diff
@@ -381,13 +382,15 @@ spdk_scsi_lun_delete(const char *lun_name)
 		spdk_scsi_dev_delete_lun(dev, lun);
 	}
 
-	/* LUNs are always created in a pair with a blockdev.
-	 * Delete the blockdev associated with this lun.
-	 */
-	spdk_bdev_unregister(lun->bdev);
+	bdev = lun->bdev;
 
 	/* Destroy this lun */
 	spdk_scsi_lun_destruct(lun);
+
+	/* LUNs are always created in a pair with a blockdev.
+	 * Delete the blockdev associated with this lun.
+	 */
+	spdk_bdev_unregister(bdev);
```

Comment:
I think this existing behavior is wrong - deleting a SCSI LUN should not delete its bdev.  Despite what the comment says, the bdev was not created by the SCSI layer, just claimed by it.  Instead, I think the bdev should just be unclaimed here.  Then the bdev could be claimed again by other LUNs created through the RPC interface.

Since spdk_scsi_lun_destruct() (which is called from this function) already unclaims the bdev, I think the simplest fix would be to just remove the spdk_bdev_unregister() call.

Does that resolve the issue that you are seeing?